### DISCUSSION TITLE
Can't stop broadcast screen to individual clients -- stops broadcast to all clients

### DISCUSSION THREAD

**Discussion by dgroos** [epoptes/epoptes#76] [Q&A]
Let's say I'm broadcasting my screen to all students' screens, but there are 3 students who no longer need what I'm sharing and need to do other work on their computer.

I naturally select just those 3 students' computers, click "stop broadcast" and unexpectedly, broadcast to all screens is halted. Then I can just reselect all the screens to whom I need to broadcast to and re-broadcast.

This workaround however is not ideal and if I release screens as students finish some written project so they can get back to their computer, well, that involves a lot of clicks and a lot of broadcasted screens disappearing... and then reapearring.

I tested this out by:
1. selecting all screens and broadcasting out to all.
2. de-selecting all screens, then right-clicking 1 screen and choosing, 'stop broadcasting'
3. unexpectedly, broadcasts to all screens stops.
Using Ubuntu Mate 18.04.1 w/ Epoptes 1.0.0-1+t201809201527~ubuntu18.04.1

**Comment by alkisg** [comment]
The menu is called "Stop broadcasts" (plural) and if you hover over the button in the toolbar it says "Stop all broadcasts on every client". So I wouldn't say its behavior is unexpected! :D

This is a nice "feature request" issue, but it's a bit complicated to implement it properly, because we rely on external VNC applications instead of using a VNC library. So we'd have to keep track of which clients are currently receiving a broadcast, and if the epoptes user ends up stopping the broadcast in all of them, at that point we'd need to terminate the "x11vnc -shared" instance running on the server.

**Comment by dgroos** [comment]
Ah! indeed, 'unexpected' is situationally defined! 
I now see it not as a bug but as a possibly nice feature to have.
Thanks!
